src/player.cpp

```diff
@@ -10186,19 +10185,14 @@ int player::get_env_resist( body_part bp ) const
         }
     }
 
-    if( bp == bp_mouth && has_bionic( bio_purifier ) && ret < 5 ) {
-        ret += 2;
-        if( ret > 5 ) {
-            ret = 5;
+    for( const bionic &bio : *my_bionics ) {
+        for( const auto &element : bio.info().env_protec ) {
+            if( element.first == bp || ( bp == bp_eyes && element.first == bp_head ) ) {
```

Comment:
Why special-case the eyes?  People can just add eyes as well if they mean it, right?